vortex-gpu/src/bit_unpack.rs

```diff
@@ -73,9 +80,65 @@ pub fn cuda_bit_unpack(
     array: &BitPackedArray,
     ctx: Arc<CudaContext>,
 ) -> VortexResult<PrimitiveArray> {
-    if array.is_empty() {
-        return Ok(Canonical::empty(array.dtype()).into_primitive());
-    }
+    let stream = ctx.default_stream();
+    let mut task = new_task(array, ctx, stream)?;
+    task.launch_task()?;
+    task.export_result().map(|c| c.into_primitive())
+}
+
+/// Returns the time (in nanoseconds) to execute just the GPU kernel, excluding memory transfers.
+/// The input array must already be allocated on the GPU.
+pub fn cuda_bit_unpack_timed(
+    array: &BitPackedArray,
+    ctx: Arc<CudaContext>,
+) -> VortexResult<Duration> {
+    let stream = ctx.default_stream();
+    let mut task = new_task(array, ctx.clone(), stream.clone())?;
+
+    let start = stream
+        .record_event(Some(CU_EVENT_DEFAULT))
+        .ok()
+        .vortex_expect("Failed to record event");
+
+    task.launch_task()?;
```

Comment:
how about we leverage the cudarc launch recording feature and return (before, after) events here?